thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -80,6 +80,9 @@ namespace diskann {
 
   template<typename T>
   PQFlashIndex<T>::~PQFlashIndex() {
+    if (this->async_generate_cache) {
```

Comment:
I think this lock is designed to ensure the thread safety of the `async` thread,  or else the `async` process may incur unpredictable behavior. However, this process may incur some risks: When the `async` task exit unexpected,  the indexes will never be released, and the release thread will be locked forever. 
Is there any way to release the `async` task directly? since the generate cache is useless for a released index. Or else, pls handle the exception in `async` task and set the flag carefully.